holoviews/core/data/xarray.py

```diff
@@ -50,9 +50,18 @@ def init(cls, eltype, data, kdims, vdims):
                 vdim = vdims[0]
             elif len(vdim_param.default) == 1:
                 vdim = vdim_param.default[0]
+                if vdim.name in data.dims:
+                    raise DataError("xarray DataArray does not define a name, "
+                                    "and the default of '%s' clashes with a "
+                                    "coordinate dimension. Give the DataArray "
+                                    "a name or supply an explicit value dimension."
+                                    % vdim.name, cls)
```

Comment:
Are you sure this works? I only see one ``%s`` and two things after ``%``. I also prefer an explicit tuple after ``%``.

Reply:
Yes, I'm sure. ``cls`` is the second argument to the error. Personally I don't like using superfluous parentheses when only one formatting argument is used.

Reply:
Ah. I favor having the parentheses in a case like this, as I don't think everyone has memorized the precedence of %.

Reply:
Ah, didn't notice this was in the context of ``DataError``. Thanks for clarifying.